package net.sf.freecol.tools;

import org.junit.*;
import static org.junit.Assert.*;

/**
 * The class <code>GenerateDocumentationTest</code> contains tests for the class <code>{@link GenerateDocumentation}</code>.
 *
 * @generatedBy CodePro at 5/10/16 11:07 AM
 * @author newsha
 * @version $Revision: 1.0 $
 */
public class GenerateDocumentationTest {
	/**
	 * Run the GenerateDocumentation() constructor test.
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_1()
		throws Exception {
		GenerateDocumentation result = new GenerateDocumentation();
		assertNotNull(result);
		// add additional test code here
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_2()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_3()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_4()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_5()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_6()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_7()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_8()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_9()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_10()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_11()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_12()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_13()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_14()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_15()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_16()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the void generateDocumentation(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGenerateDocumentation_17()
		throws Exception {
		String[] languages = new String[] {};

		GenerateDocumentation.generateDocumentation(languages);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at net.sf.freecol.tools.GenerateDocumentation.generateDocumentation(GenerateDocumentation.java:172)
	}

	/**
	 * Run the String getResource(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testGetResource_1()
		throws Exception {
		String key = "";

		String result = GenerateDocumentation.getResource(key);

		// add additional test code here
		assertEquals(null, result);
	}

	/**
	 * Run the String localize(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testLocalize_1()
		throws Exception {
		String template = "";

		String result = GenerateDocumentation.localize(template);

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String localize(String,String,String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test
	public void testLocalize_2()
		throws Exception {
		String template = "";
		String key = "";
		String number = "";

		String result = GenerateDocumentation.localize(template, key, number);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NumberFormatException: empty String
		//       at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1842)
		//       at sun.misc.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
		//       at java.lang.Double.parseDouble(Double.java:538)
		//       at net.sf.freecol.tools.GenerateDocumentation.localize(GenerateDocumentation.java:227)
		assertNotNull(result);
	}

	/**
	 * Run the String localize(String,String,String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test(expected = java.lang.NumberFormatException.class)
	public void testLocalize_3()
		throws Exception {
		String template = "";
		String key = "";
		String number = "";

		String result = GenerateDocumentation.localize(template, key, number);

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the void main(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test(expected = java.lang.NullPointerException.class)
	public void testMain_1()
		throws Exception {
		String[] args = new String[] {null};

		GenerateDocumentation.main(args);

		// add additional test code here
	}

	/**
	 * Run the void main(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Test(expected = java.lang.NullPointerException.class)
	public void testMain_2()
		throws Exception {
		String[] args = new String[] {};

		GenerateDocumentation.main(args);

		// add additional test code here
	}

	/**
	 * Perform pre-test initialization.
	 *
	 * @throws Exception
	 *         if the initialization fails for some reason
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@Before
	public void setUp()
		throws Exception {
		// add additional set up code here
	}

	/**
	 * Perform post-test clean-up.
	 *
	 * @throws Exception
	 *         if the clean-up fails for some reason
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	@After
	public void tearDown()
		throws Exception {
		// Add additional tear down code here
	}

	/**
	 * Launch the test.
	 *
	 * @param args the command line arguments
	 *
	 * @generatedBy CodePro at 5/10/16 11:07 AM
	 */
	public static void main(String[] args) {
		new org.junit.runner.JUnitCore().run(GenerateDocumentationTest.class);
	}
}